nemo/collections/llm/gpt/model/megatron/hyena/hyena_utils.py

```diff
@@ -561,13 +561,22 @@ def get_t(self, L: int) -> torch.Tensor:
             self.t = t
             return self.t
 
-    def get_logp(self):
+    def get_logp(self, context_parallel_group=None):
         """Compute the log poles for the implicit modal filter."""
-        logp = -torch.exp(self.p.to(torch.float32))
-        glogp = logp * torch.exp(self.gamma.to(torch.float32))
+        if context_parallel_group is not None:
```

Comment:
should this be `and context_parallel_group.size() > 1:`?

Reply:
I see your point, if `_hyena_use_cp` set to False, this can be None. Slightly confusing code, b/c it sets cp_group to None.